cmd/sidecars/smbios/smbios.go

```diff
@@ -40,12 +40,12 @@ const (
 func onDefineDomain(vmiJSON, domainXML []byte) (string, error) {
 	vmiSpec := vmSchema.VirtualMachineInstance{}
 	if err := json.Unmarshal(vmiJSON, &vmiSpec); err != nil {
-		return "", err
+		return "", fmt.Errorf("Failed to unmarshal given VMI spec: %s %s", err, string(vmiJSON))
 	}
 
 	domainSpec := api.DomainSpec{}
 	if err := xml.Unmarshal(domainXML, &domainSpec); err != nil {
-		return "", err
+		return "", fmt.Errorf("Failed to unmarshal given Domain spec: %s %s", err, string(domainXML))
```

Comment:
I remember I had [issues](https://github.com/kubevirt/kubevirt/pull/3856) with logging the XML, therefore I had to decode it, did the log worked for you in this case?

Reply:
I haven't tested this. I've added the conditional in case of errors, we have something to help debug.
The domainXML should be a valid xml string, no sure what you mean by having to decode it.